code/modules/reagents/reagent_containers.dm

```diff
@@ -122,9 +124,40 @@
 
 	reagents.clear_reagents()
 
+//melts plastic beakers
 /obj/item/reagent_containers/microwave_act(obj/machinery/microwave/M)
 	reagents.expose_temperature(1000)
+	if(temp_immune == TRUE)

```

Comment:
put the added code in a "if(!temp_immune)" rather than doing this, because you just made it skip over the ..() call.
same thing for the other procs.

Reply:
Done